Scale factor-related fixes to wxImageList and Mac wxDataViewCtrl

vadz opened the pull request:
This fixes various issues reported in #22189 (and also adds an "Exit" Tango bitmap just to make the popup menu in the menu sample a bit less ugly in high DPI but this is not the important part).

@TcT2k I've restored the default size of the Tango bitmaps removed in your 73d37d460df4d0c84422fa4f7e300e373f62da80 because I'm not sure if this was done intentionally and, as this changes the behaviour of the existing code, I'd prefer to avoid changing this unless there is some good reason to do it -- please let me know if there is.

@a-wi I've reverted your b6d305e4f24aaaa8ffac561d8d119176cb0e9707 for the reasons discussed on wx-dev and in the issue and briefly summarized in the commit message. Please let me know if this change (re)breaks something but we're going to need to fix the real problem, whichever it is, in some other way because just silently dropping the images is not a good idea.

vadz committed:
Add wxART_QUIT icon to Tango art provider

Gnome uses a variant of the "log out" icon for its standard "Exit" icon,
so it should be fine for us to do it too, contrary to what the comment
from back when Tango provider was added in c1d2466a79 (Add wxArtProvider
using Tango icons., 2010-12-31) was saying.

vadz committed:
Restore the old default size of Tango art provider bitmaps

The default size of the bitmaps created from bitmap bundles returned by
the Tango art provider for the client strings different from wxART_MENU
and wxART_BUTTON was changed from 24x24 to 16x16 in 766fefeb34 (Use SVG
instead of PNGs in tango art provider, 2022-01-27), see #22067, without
any real need and perhaps even non intentionally at all.

Restore the old default size to restore the old behaviour.

See #22189.

vadz committed:
Revert "Assure that all images in the generic wxImageList have the same scale factor"

This reverts commit b6d305e4f24aaaa8ffac561d8d119176cb0e9707 because
silently discarding images using a different scale factor is simply too
user-unfriendly: this can silently break the existing code, i.e. it can
still compile perfectly and not give any errors during run-time but not
show any images neither.

See #22189.

vadz committed:
Use appropriately sized bitmaps in wxOSX wxDataViewCtrl

Select the bitmap of the required size from wxBitmapBundle instead of
selecting the bitmap with the default size and then possibly rescaling
it to get the bitmap of the needed size, resulting in fuzzy-looking
bitmaps.

Closes #22189.

dkulp commented:
Now I'm not getting any bitmaps showing in me wxTreeListCtrl.    I'll try and dig in an see what I can find later today.

<img width="202" alt="Screen Shot 2022-03-21 at 2 10 38 AM" src="https://user-images.githubusercontent.com/91569/159261521-92a57e46-7738-46a5-bc33-b82f7ca3fa90.png">

vadz committed:
fixup! Restore the old default size of Tango art provider bitmaps

vadz commented:
This could happen if the width of the bitmaps being added doesn't match (is smaller than) the width specified when creating the `wxImageList`. Is it possible that you create it with the size 32x32 and then add 32x32@2x images to it, so that their actual size is 16x16 and so `wxGenericImageList::Add()` just throws them away?

vadz committed:
squash! Revert "Assure that all images in the generic wxImageList have the same scale factor"

Also update the tests reverting most of dc43d15cf7

vadz committed:
Update wxImageList documentation to mention high DPI concerns

First of all, document that this class is (informally) deprecated.

Also document that the various sizes are expressed in logical pixels.

vadz commented:
I'm going to merge this soon if there are no objections.
